### PR TITLE
Task NXP-30153 improve pipeline build/stage result

### PR DESCRIPTION
**Pull request by kevinleturc**

### CODE REVIEW

File: ci/Jenkinsfiles/build.groovy [hunk 1]
```diff
@@ -527,7 +527,7 @@ pipeline {
           ----------------------------------------"""
           runFunctionalTests('ftests')
         }
-        findText regexp: ".*ERROR.*", fileSet: "ftests/**/log/server.log"
+        findText regexp: ".*ERROR.*", fileSet: "ftests/**/log/server.log", buildResult: 'UNSTABLE'
```
Comment (atchertchian) [inline ci/Jenkinsfiles/build.groovy:530]: Maybe adapt the slack notif accordingly (?) see https://github.com/nuxeo/nuxeo-explorer/blob/master/ci/Jenkinsfiles/build.groovy#L412 (where ERROR in logs are frequent too)

**Comment by kevinleturc** [comment]
Trying another approach by declaring the plugin like mentioned [here](https://plugins.jenkins.io/text-finder/)

**Comment by kevinleturc** [comment]
Trying another approach by putting findText in the post/always step.

File: ci/Jenkinsfiles/build.groovy (hunk 1 above)
Comment (kevinleturc) [reply ci/Jenkinsfiles/build.groovy]: This change was more a test to help me to understand what is going on. In the end, this stage should make the test FAILURE, the change has been removed.

File: ci/Jenkinsfiles/build.groovy [hunk 2]
```diff
@@ -262,7 +260,7 @@ def buildUnitTestStage(env) {
 }
 
 void archiveKafkaLogs(namespace, logFile) {
-  sh "kubectl logs ${TEST_KAFKA_POD_NAME} --namespace=${namespace} > ${logFile}"
+  sh "kubectl logs ${TEST_KAFKA_POD_NAME} --namespace=${namespace} > ${logFile} || echo"
```
Comment (efge) [inline ci/Jenkinsfiles/build.groovy:263]: What's the `|| echo` for? If it's to not stop if the previous command returns an error, the traditional way is `|| true`

Comment (kevinleturc) [reply ci/Jenkinsfiles/build.groovy:263]: It is to avoid the error indeed, I think I tried the echo way without success, let do another try 🤞.

Comment (efge) [reply ci/Jenkinsfiles/build.groovy:263]: `echo` and `true` both returns a status 0 (ok) so they should be equivalent in terms of error recovery, it's just that there's no need to output a newline.

**Review by troger — APPROVED** [review]

**Review by ataillefer — APPROVED** [review]

**Review by efge — APPROVED** [review]

**Pull request merged**